stdlib/public/Concurrency/ThreadSanitizer.cpp

```diff
@@ -55,3 +55,9 @@ void swift::_swift_tsan_release(void *addr) {
     tsan_release(addr);
   }
 }
+
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(c)
+void swift_task_set_tsan_hooks(TSanFunc *acquire, TSanFunc *release) {
+  tsan_acquire = acquire;
+  tsan_release = release;
+}
```

Comment:
Adopting this will be sequenced:
  1. Call this hook from the TSan runtime
  2. Remove `dlsym()` lookup from Swift runtime in a follow-up

```
➤ nm -g build/Ninja-RelWithDebInfoAssert+stdlib-DebugAssert/swift-macosx-x86_64//lib/swift/macosx/libswift_Concurrency.dylib | grep tsan
0000000000057a40 T _swift_task_set_tsan_hooks
```

Let me know if you have any remaining concerns.
@mikeash @kubamracek @devincoughlin

Reply:
rdar://74256720 (Call swift_task_set_tsan_hooks() from TSan runtime)
rdar://75493372 (Remove dlsym() lookup for TSan functions)

Reply:
Sounds fine to me, thanks! All LGTM.